realm/realm-library/src/main/java/io/realm/internal/android/AndroidCapabilities.java

```diff
@@ -50,6 +50,15 @@ public void checkCanDeliverNotification(String exceptionMessage) {
         }
     }
 
+    @Override
+    public boolean isMainThread() {
+        return looper == Looper.getMainLooper();
```

Comment:
`looper != null && looper == Looper.getMainLooper()` (if `getMainLooper()` is expensive)